sage/embedder.py

```diff
@@ -52,15 +52,23 @@ def __init__(
 
     def embed_dataset(self, chunks_per_batch: int, max_embedding_jobs: int = None) -> str:
         """Issues batch embedding jobs for the entire dataset. Returns the filename containing the job IDs."""
+        total_chunks = 0
+        for content, metadata in self.data_manager.walk():
+            chunks = self.chunker.chunk(content, metadata)
+            total_chunks += len(chunks)
```

Comment:
Walking over the entire dataset and chunking it is an expensive operation, so we shouldn't be doing it twice (here and on line 67) just so that we have the total number of chunks in advance.

A less compute-intensive alternative would be to use the number of files. Something like:
```
num_files = len([x for x in self.data_manager.walk(get_content=False)])
pbar = tqdm(total=num_files, ...)

for content, metadata in self.data_manager.walk():
   ...
   pbar.update(1)
```

Reply:
So should i initialize the progress bar with the no. of files, instead of chunks?

Reply:
Yes, exactly.

Reply:
cool, then i am applying the same changes in all the embedders